Assets/Code/Interactables/Effects/Static Effects/Model/AttachedStaticEffect.cs

```diff
@@ -0,0 +1,11 @@
+﻿public abstract class AttachedStaticEffect : StaticEffect {
+    // This StaticEffect child class deals specifically with effects that are attached to specific entities
+    // If a character has a modifier or status applied to it, it will be an instance of this class
+
+    // reference to the character that has the static effect
+    public Fighter character;
```

Comment:
again, probably need to make an IAttachable interface or Character class or something at a higher level than Fighter.. also, isn't this a dependency loop? shouldn't it only be defined whether the fighet has an attachment or whether the attachment is assigned to a fighter, not both?

Reply:
it is not so much a dependency loop as it is a 2 way relationship. Fighters can have many attached effects, but an effect can only have one fighter. When operating in the scope of the effect class, it's important to have access to the fighter's data because of how effects are structured. Even though it's redundant, it allows for checking effects based on individual fighters quickly rather than having to go through the StaticEffectController which just controls a running list of literally every active effect.

Reply:
instead of having a field, can this be implemented as a method then? getAttachedFighter() (temp name) ?